src/planner/rules/expr.rs

```diff
@@ -118,7 +118,7 @@ pub fn eval_constant(egraph: &EGraph, enode: &Expr) -> ConstValue {
         Some(x(*e)?.clone())
     } else if let Some((op, a, b)) = enode.binary_op() {
         let (a, b) = (x(a)?, x(b)?);
-        if a.is_null() || b.is_null() {
+        if a.is_null() || b.is_null() || b.is_zero() {
```

Comment:
should be `b.is_zero() && op == Div`. otherwise any `a + 0` or `a - 0` would result in NULL.
However, I believe this condition is unnecessary if array division is implemented correctly.